web/static/js/components/layout/SaveStatus.jsx

```diff
@@ -3,17 +3,42 @@ import { connect } from 'react-redux'
 import TimeAgo from 'react-timeago'
 import { withRouter } from 'react-router'
 
+import { translate } from 'react-i18next'
+
 class SaveStatus extends Component {
-  static propTypes = {
-    saveStatus: PropTypes.any,
-    routes: PropTypes.any
+
+  formatter(number, unit, suffix, date, defaultFormatter) {
+    const { t } = this.props
+
+    if (unit == 'second') {
+      return t('All changes saved')
+    }
+
+    switch (unit) {
+      case 'minute':
+        return t(`Last saved {{count}} minute ago`, {count: number})
+      case 'hour':
+        return t(`Last saved {{count}} hour ago`, {count: number})
+      case 'day':
+        return t(`Last saved {{count}} day ago`, {count: number})
+      case 'week':
+        return t(`Last saved {{count}} week ago`, {count: number})
+      case 'month':
+        return t(`Last saved {{count}} month ago`, {count: number})
+      case 'year':
+        return t(`Last saved {{count}} year ago`, {count: number})
+      default:
+        return t(`Last saved {{count}} ${unit} {{text}}`, {count: number, text: suffix})
+    }
   }
 
   render() {
-    const { saveStatus, routes } = this.props
+    const { saveStatus, routes, t } = this.props
 
     let show = false
 
+    const bindedFormatter = this.formatter.bind(this)
```

Comment:
Can we bind this in the constructor? to avoid having to do it in every render cycle